recipe/build.sh

```diff
@@ -1,18 +1,47 @@
 #!/bin/bash
+set -x
 
 IFS='.' read -r -a PKG_VER_ARRAY <<< "${PKG_VERSION}"
 
 sed -i.bak "s/libLTO.dylib/libLTO.${PKG_VER_ARRAY[0]}.dylib/g" lib/Driver/ToolChains/Darwin.cpp
 
-mkdir build
-cd build
-
 if [[ "$variant" == "hcc" ]]; then
-  EXTRA_ARGS="$EXTRA_ARGS -DKALMAR_BACKEND=HCC_BACKEND_AMDGPU -DHCC_VERSION_STRING=2.7-19365-24e69cd8-24e69cd8-24e69cd8"
-  EXTRA_ARGS="$EXTRA_ARGS -DHCC_VERSION_MAJOR=2 -DHCC_VERSION_MINOR=7 -DHCC_VERSION_PATCH=19365"
-  EXTRA_ARGS="$EXTRA_ARGS -DKALMAR_SDK_COMMIT=24e69cd8 -DKALMAR_FRONTEND_COMMIT=24e69cd8 -DKALMAR_BACKEND_COMMIT=24e69cd8"
+  CMAKE_ARGS="$CMAKE_ARGS -DKALMAR_BACKEND=HCC_BACKEND_AMDGPU -DHCC_VERSION_STRING=2.7-19365-24e69cd8-24e69cd8-24e69cd8"
+  CMAKE_ARGS="$CMAKE_ARGS -DHCC_VERSION_MAJOR=2 -DHCC_VERSION_MINOR=7 -DHCC_VERSION_PATCH=19365"
+  CMAKE_ARGS="$CMAKE_ARGS -DKALMAR_SDK_COMMIT=24e69cd8 -DKALMAR_FRONTEND_COMMIT=24e69cd8 -DKALMAR_BACKEND_COMMIT=24e69cd8"
+fi
+
+if [[ "$variant" == "root"* ]]; then
+  # Cling needs some minor patches to the LLVM sources
+  sed -i "s@LLVM_LINK_LLVM_DYLIB yes@LLVM_LINK_LLVM_DYLIB no@g" "${PREFIX}/lib/cmake/llvm/LLVMConfig.cmake"
```

Comment:
Why?

Reply:
I don’t remember the exact details but it was something along the lines of dynamically linking Clang to LLVM causes Cling to fail on launch with an error like this:

```
: CommandLine Error: Option 'disable-symbolication' registered more than once!
LLVM ERROR: inconsistency in registered CommandLine options
++ printf '\033]0;%s@%s:%s\007' conda 6b92b93499a2 '~/feedstock_root/build_artifacts/debug_1615404201085/work/build-dir'
```

Reply:
I couldn’t find a less ugly workaround but I’m open to one if you have ideas

Reply:
Is cling statically linking to LLVM?

Reply:
Yes and its not possible to use it without doing so.

Reply:
Is cling linking to static clang libraries? Is it using `-Wl,--exclude-libs,ALL`?

Reply:
Cling statically links to parts of both LLVM and Clang. Dynamically linking to either causes a bunch of issues though I’d have to ask upstream to remember the details as I’ve forgotten. The symbols are all hidden though so it doesn’t interfere with other packages that use LLVM/Clang.

Reply:
@isuruf Do you have any other thoughts on this PR?

Reply:
If cling statically links to parts of both LLVM and Clang, why does the clang tools need to link to LLVM statically. I don't understand.

Reply:
I'll build again without this line to try to remind myself of the issue and get back to you

Reply:
Removing this line and leaving `LLVM_LINK_LLVM_DYLIB=yes` causes the build of `clangdev` to fail with:

```bash
+ clang-9 --version
: CommandLine Error: Option 'use-dbg-addr' registered more than once!
LLVM ERROR: inconsistency in registered CommandLine options
Tests failed for clang-9-9.0.1-root_62400_h12806e0_10000.conda - moving package to /home/cburr/mambaforge/conda-bld/broken
```

I don't really understand why, I guess something in the patches needed for root/cling is buggy but glancing through I have no idea what it would be.

Reply:
Actually I think that might be been caused by my environment getting into a bad state with all of the various builds I'd made. The big issue is that the CMake targets are invalid when clang dynamically links to LLVM. i.e. in `lib/cmake/clang/ClangTargets.cmake`:

With `LLVM_LINK_LLVM_DYLIB=yes` the shared `LLVM` target is listed as a dependency:

```cmake
168:set_target_properties(clangFrontend PROPERTIES
169-  INTERFACE_LINK_LIBRARIES "clangAST;clangBasic;clangDriver;clangEdit;clangLex;clangParse;clangSema;clangSerialization;LLVM"
170-)
```

With `LLVM_LINK_LLVM_DYLIB=no` the static libraries are listed as dependencies:
```
168:set_target_properties(clangFrontend PROPERTIES
169-  INTERFACE_LINK_LIBRARIES "clangAST;clangBasic;clangDriver;clangEdit;clangLex;clangParse;clangSema;clangSerialization;LLVMBitReader;LLVMBitstreamReader;LLVMOption;LLVMProfileData;LLVMSupport"
170-)
```

Reply:
@isuruf Any objections to merging?

Reply:
Will do. Thanks for reviewing! 🍰